docs/proposals/06-storage-encryption.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Storage Encryption
+
+oep-number: 6
+
+creation-date: 2023-01-03
+
+status: implementable
+
+authors:
+- "@Ashughorla"
+- "@kasabe28"
+- "@ushabelgur"
+  
+reviewers:
+- "@ManuStoessel"
+- "@adracus"
+- "@afritzler"
+- "@lukasfrank"
+
+---
+
+# OEP-6: Storage Encryption
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Details](#Details)
+- [Proposal](#proposal)
+
+## Summary
+One of the important feature of Cloud Native IaaS is to provide secure storage. This proposal focuses on providing option to enable encryption for individual onmetal Volume.
+
+## Motivation
+As part of Storage encryption feature Onmetal API to support option to enable encryption of Volumes. Each `Volume` can set encryption enabled flag and provide encryption key secret reference(Optional).
+
+### Goals
+  - Allow user to enable/disable volume encryption with flag
+  - Encrypt volume with user provided encryption key
+
+### Details
+  - By default encryption enabled flag will be set to false
+  - If encryption enabled flag is set to true, then user can provide encryption key via secret reference to encrypt onmetal `Volume`.
+  - If encryption enabled flag is set to true and user has not provided encryption key secret reference in `Volume`, then `Volume` will be encrypted with shared key of storage provider.
```

Comment:
should we really default to using the shared key of the storage provider or should we create a secret with a new random key for the user? Similar to the way the user would create the secret themselves we could auto-create a secret with a predictable name (e.g. same name as the `Volume`) and fill the `encryption.keySecretRef` with the name of the auto-generated secret.

Reply:
The storage provider does not and should not offer a shared key.
We should have the encryption secret mandatory in the API (renaming to `secretRef` would be more concise & still imply the same use).
Default keys and other QoL use cases can be implemented as separate (admission / Kyverno etc.) plugins and should not be part of the API extension proposal.

Reply:
> We should have the encryption secret mandatory in the API

Only when encryption enabled is set to true?

Reply:
If we don't want to use any default key as of now, then we can remove encryption enabled flag and make `secretRef ` as optional